File: szakdoliV2/Drawable/Polygon.py
# -*- coding: utf-8 -*-

import pygame

from Base import AbstractDrawable
from System.Constants import COLOR as Color

class Polygon(AbstractDrawable):

    def __init__(self, cooridantes, color=Color.BLACK, width = 0):
        self.coordinates = cooridantes
        self.color = color
        self.width = width
        self.accentColor = Color.WHITE
        self.transparent = False
        self.base = { "coords" : self.coordinates }

    def DrawObject(self, screen):
        if not self.transparent:
            pygame.draw.polygon(screen, self.accentColor, self.coordinates, 0)
        pygame.draw.polygon(screen, self.color, self.coordinates, self.width)

    def IsInside(self, position):
        pass

    def ResetPosition(self):
        self.coordinates = self.base["coords"]

    def GetParameters(self):
        pass

File: docs/Drawable/Polygon.js
import { AbstractDrawable } from "./Base/AbstractDrawable.js";
import { COLOR } from "../System/Constants.js";
import { DrawPolygon } from "../System/SupportFunctions.js";

export class Polygon extends AbstractDrawable{

    constructor(coordinates, color=COLOR.BLACK, width=0) {
        super();
        this.coordinates = coordinates;
        this.color = color;
        this.width = width;
        this.accentColor = COLOR.WHITE;
        this.transparent = false;
        this.base = { "coords" : this.coordinates };
    }

    DrawObject(screen) {
        if(!this.transparent){
            DrawPolygon(screen, this.accentColor, this.coordinates);
        }
        DrawPolygon(screen, this.color, this.coordinates, this.width);
    }

    IsInside(position) {
    }

    ResetPosition(){
        this.coordinates = this.base.coordinates;
    }

    GetParameters(){
    }
}